test/endpoints.test.js

```diff
@@ -0,0 +1,170 @@
+        const response = await request(app)
+            .post("/api/allocations")
+            .send(requestData)
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("NFT Info Endpoint", () => {
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("post /api/nftinfo should return NFT info data", async () => {
+        const response = await request(app).post("/api/nftinfo")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("NFT Vols Endpoint", () => {
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("post /api/volume should return NFT volumes data", async () => {
+        const response = await request(app).post("/api/volume")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("Rewards Info Endpoint", () => {
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("post /api/rewards should return rewards info data", async () => {
+        const response = await request(app).post("/api/rewards")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("Rewards Summary Endpoint", () => {
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("post /api/rewardsSummary should return rewards summary data", async () => {
+        const response = await request(app).post("/api/rewardsSummary")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("vebals Endpoint", () => {
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("post /api/vebals should return vebals data", async () => {
+        const response = await request(app).post("/api/vebals")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual(mockResponse)
+    })
+})
+
+describe("APY Endpoint", () => {
+    const { getActiveApy, getApyByAddr } = require("../src/services/apy")
+    jest.mock("../src/services/apy", () => ({
+        getActiveApy: jest.fn(),
+        getApyByAddr: jest.fn()
+    }))
+
+    beforeEach(() => {
+        jest.clearAllMocks()
+    })
+
+    test("get /api/apy/active should return active apy", async () => {
+        getActiveApy.mockResolvedValueOnce(0)
+        const response = await request(app).get("/api/apy/active")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual({ apy: 0 })
+    })
+
+    test.skip("get /api/apy/active should return active apy", async () => {
+        getApyByAddr.mockImplementation((query, values, callback) => {
+            callback(null, 42)
+        })
+
+        const response = await request(app).get("/api/apy/addr/test")
+
+        expect(response.status).toBe(200)
+        expect(response.body).toEqual({ apy: 0 })
```

Comment:
here, it would seem to me like the mock implementation would return 42
```
       getApyByAddr.mockImplementation((query, values, callback) => {
            callback(null, 42)
        })
```

when calling /addr/test
```
        const response = await request(app).get("/api/apy/addr/test")
```

which means that apy should equal to 42, no?
```
router.get("/addr/:addr", async (req, res) => {
    let apy = await getApyByAddr(req.params.addr)
})
```

But I have to assume it's returning null, undefined, since the final answer is 0
```
    res.json({ apy: apy ?? 0 })
```

Will run tests locally and step through this.